src/bijectors/corr.jl

```diff
@@ -205,30 +222,34 @@ julia> y = b(X)  # Transform to unconstrained vector representation.
 julia> inverse(b)(y) ≈ X  # (✓) Round-trip through `b` and its inverse.
 true
 """
-struct VecCorrBijector <: Bijector end
-with_logabsdet_jacobian(b::VecCorrBijector, x) = transform(b, x), logabsdetjac(b, x)
+struct VecCorrBijector <: AbstractVecCorrBijector end
+transform(::Inverse{VecCorrBijector}, y::AbstractVector{<:Real}) = pd_from_upper(_inv_link_chol_lkj(y))
 
-function transform(::VecCorrBijector, X::AbstractMatrix{<:Real})
-    w = upper_triangular(parent(cholesky(X).U))
-    r = _link_chol_lkj(w)
+logabsdetjac(::Inverse{VecCorrBijector}, y::AbstractVector{<:Real}) = _logabsdetjac_inv_corr(y)
 
-    # Extract only the upper triangle of `r`.
-    return triu1_to_vec(r)
-end
+struct VecTriuBijector <: AbstractVecCorrBijector end
 
-function transform(::Inverse{VecCorrBijector}, y::AbstractVector{<:Real})
-    Y = vec_to_triu1(y)
-    w = _inv_link_chol_lkj(Y)
-    return pd_from_upper(w)
+function transform(::Inverse{VecTriuBijector}, y::AbstractVector{<:Real})
+    # This constructor is compatible with Julia v1.6
+    # for later versions Cholesky(::UpperTriangular) works
+    U = UpperTriangular(_inv_link_chol_lkj(y))
+    return Cholesky(U.data, 'U', 0)
 end
 
-function logabsdetjac(b::VecCorrBijector, x)
-    return -logabsdetjac(inverse(b), b(x))
-end
-function logabsdetjac(::Inverse{VecCorrBijector}, y::AbstractVector{<:Real})
-    return _logabsdetjac_chol_lkj(vec_to_triu1(y))
+logabsdetjac(::Inverse{VecTriuBijector}, y::AbstractVector{<:Real}) = _logabsdetjac_inv_chol(y)
+
+struct VecTrilBijector <: AbstractVecCorrBijector end
+
+function transform(::Inverse{VecTrilBijector}, y::AbstractVector{<:Real})
+    # This constructor is compatible with Julia v1.6
+    # for later versions Cholesky(::LowerTriangular) works
+    L = LowerTriangular(_transpose_matrix(_inv_link_chol_lkj(y)))
```

Comment:
```suggestion
    # HACK: Need to make materialize the transposed matrix to avoid numerical instabilities.
    # If we don't, the return-type can be both `Matrix` and `Transposed`.
    L = LowerTriangular(Matrix(transpose(_inv_link_chol_lkj(y))))
```